MusicCloud/MusicCloud/Scenes/HomeViewController.swift

```diff
@@ -9,14 +9,166 @@
+            return APIConstant.DUBSTEP
+        case .Indie:
+            return APIConstant.INDIE
+        case .Rock:
+            return APIConstant.ROCK
+        case .Pop:
+            return APIConstant.POP
+        case .House:
+            return APIConstant.HOUSE
+        case .Trap:
+            return APIConstant.TRAP
+        case .Trance:
+            return APIConstant.TRANCE
+        case .Classical:
+            return APIConstant.CLASSICAL
+        }
+    }
+}
+
 final class HomeViewController: UIViewController {
+    @IBOutlet private weak var tableView: UITableView!
+    private var trackRepository = TrackRepository(api: APIService.share)
+    private var trackList: [Genre: [Track]] = [Genre.Disco: [Track](),
+                                               Genre.Rock: [Track](),
+                                               Genre.Pop: [Track](),
+                                               Genre.Dubstep: [Track](),
+                                               Genre.Indie:[Track](),
+                                               Genre.Country: [Track](),
+                                               Genre.House: [Track](),
+                                               Genre.Trap: [Track](),
+                                               Genre.Trance: [Track](),
+                                               Genre.Classical: [Track]()]
+    private var limit = 20
+    private let refreshControl = UIRefreshControl()
+    private var storedOffsets = [Int: CGFloat]()
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         setupViews()
+        getData()
     }
     
     private func setupViews() {
         navigationItem.title = NavigationConstant.homeNavItemTitle
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(cellType: TableViewCell.self)
+        if #available(iOS 10.0, *) {
+            tableView.refreshControl = refreshControl
+        } else {
+            tableView.addSubview(refreshControl)
+        }
+        refreshControl.addTarget(self, action: #selector(reloadTableView), for: .valueChanged)
+    }
+    
+    @objc private func reloadTableView() {
+        trackList.keys.forEach {
+            trackList[$0] = [Track]()
+        }
+        getData()
+        refreshControl.endRefreshing()
+    }
+    
+    private func getData() {
+        getDataByGenre(genre: Genre.Disco)
+        getDataByGenre(genre: Genre.Rock)
+        getDataByGenre(genre: Genre.Pop)
+        getDataByGenre(genre: Genre.Dubstep)
+        getDataByGenre(genre: Genre.Indie)
+        getDataByGenre(genre: Genre.Country)
+        getDataByGenre(genre: Genre.House)
+        getDataByGenre(genre: Genre.Trap)
+        getDataByGenre(genre: Genre.Trance)
+        getDataByGenre(genre: Genre.Classical)
+    }
+    
+    private func getDataByGenre(genre: Genre) {
+        trackRepository.getTrackByGenre(genre: genre.createGenreParam(), limit: limit) { (response) in
```

Comment:
```suggestion
        trackRepository.getTrackByGenre(genre: genre.createGenreParam(), limit: limit) { [weak self] response in
```
đoạn này nếu ko có weak self sẽ có thể tạo ra leak memory